### PR TITLE
fix: newline-between-instructions false positive with comment blank lines

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Fixes a false positive in `tally/newline-between-instructions` (grouped mode) when a comment attached to the next instruction has a blank line between it and the instruction
- The gap was computed using only the distance from the previous instruction to the comment's effective start, missing blank lines *within* the comment block that still provide visual separation
- Applied the same fix to the fix resolver so `--fix` doesn't insert unnecessary blank lines

### Example (no longer flagged)

```dockerfile
# a comment
FROM aws/lambda/python:3.12
# some comment

ENV PYTHONUNBUFFERED=1 \
  PYTHONDONTWRITEBYTECODE=1
```

## Test plan

- [x] New integration fixture `newline-between-instructions-comment-gap` with `wantExit: 0`
- [x] All existing `newline-between-instructions` lint and fix tests pass
- [x] Full test suite passes with `-race`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of blank lines that appear inside preceding comment blocks so they correctly count as visual separation between instructions, reducing incorrect spacing warnings and preventing unnecessary newline violations.

* **Tests**
  * Added an integration test case and snapshot for newline-between-instructions scenarios involving comment gaps, plus unit tests covering blank-line detection to ensure correct behavior across edge cases.

### CODE REVIEW

File: internal/sourcemap/sourcemap.go [hunk 1]
```diff
@@ -271,6 +271,17 @@ func (sm *SourceMap) EffectiveStartLine(startLine int, prevComments []string) in
...
+// HasBlankLineBetween reports whether any blank line exists strictly between
+// startLine and endLine (both 1-based, exclusive on both ends).
+func (sm *SourceMap) HasBlankLineBetween(startLine, endLine int) bool {
+	for line := startLine + 1; line < endLine; line++ {
+		if strings.TrimSpace(sm.Line(line-1)) == "" {
+			return true
+		}
+	}
+	return false
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/sourcemap/sourcemap.go]: ✅ Addressed in ea08fd9. Clamped bounds (previous commit) and went further — extracted the entire duplicated gap+mode+comment logic into `SourceMap.ComputeNewlineGap`, eliminating the CPD duplicate entirely (-83 lines, +66 lines). Thanks @coderabbitai\!
